core/include/crypto/crypto_secure_element.h

```diff
@@ -0,0 +1,44 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) Foundries Ltd. 2021 - All Rights Reserved
+ * Author: Jorge Ramirez <jorge@foundries.io>
+ */
+
+ /*
+  * This is the Cryptographic Secure Element API, part of the Cryptographic
+  * Provider API.
+  *
+  * These requests shall be handled in the secure element normally placed on
+  * a serial communication bus (SPI, I2C).
+  */
+#ifndef __CRYPTO_SECURE_ELEMENT_H
+#define __CRYPTO_SECURE_ELEMENT_H
+
+#include <tee_api_types.h>
+
+#if defined(CFG_NXP_SE05X)
```

Comment:
the functions are not SE05x specific. remove this directive.

Reply:
yeah, well is the only supported SE...but ok, do you want me to continue adding functionality and squashing? the list is growing...